pkg/kapp/diff/prepare_res_with_periodic_ann.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
+)
+
+const (
+	MaxDurationAnnKey   = "kapp.k14s.io/max-duration"
+	lastRenewTimeAnnKey = "kapp.k14s.io/last-renewed-time"
+)
+
+func CheckAndCalculateResWithPeriodicAnn(existingResources, newResources []ctlres.Resource) error {
+	var (
+		duration time.Duration
+		err      error
+	)
+
+	addNonceMod := ctlres.StringMapAppendMod{
+		ResourceMatcher: ctlres.AllMatcher{},
+		Path:            ctlres.NewPathFromStrings([]string{"metadata", "annotations"}),
+		KVs: map[string]string{
+			lastRenewTimeAnnKey: fmt.Sprintf("%v", time.Now().UTC().Format(time.RFC3339)),
+		},
+	}
+	existRes := existResMap(existingResources)
+
+	for _, res := range newResources {
+		rs, found := existRes[res.Name()+res.Kind()]
+		if found {
+			val, exist := rs.Annotations()[MaxDurationAnnKey]
+			if exist {
+				duration, err = time.ParseDuration(val)
```

Comment:
We should use `res` as opposed to `val` here

Something @cppforlife pointed out ^

Reply:
Closing this and all other outdated comments as changes has been updated with new implementation logic.